oak/src/oak/server.cpp

```diff
@@ -175,75 +112,207 @@ class Request : public cloe::Request {
 
  public:
   /**
-   * Create a new Request from ServerImpl.
-   *
-   * - This should not panic.
+   * Creates a new Request from a http_connection
    */
-  explicit Request(const ServerImpl::request& req, const ServerImpl::connection_ptr& conn)
-      : req_(req)
-      , conn_(conn)
-      , endpoint_(Muxer<Handler>::normalize(req.destination))
-      , queries_(parse_queries(req.destination)) {
-    from_string(req.method, method_);
-    if (method_ == RequestMethod::POST) {
-      body_ = DataReceiver<ServerImpl>(req, conn).read_all();
-    }
-  }
+  explicit Request(const http_connection& connection);
 
   RequestMethod method() const override { return method_; }
   ContentType type() const override { return ContentType::UNKNOWN; }
   const std::string& body() const override { return body_; }
-  const std::string& uri() const override { return req_.destination; }
+  const std::string& uri() const override { return uri_; }
   const std::string& endpoint() const override { return endpoint_; }
   const std::map<std::string, std::string>& query_map() const override { return queries_; }
 };
 
-ServerImplHandler::ServerImplHandler() {
-  muxer.set_default([this](const cloe::Request& q, cloe::Response& r) {
-    ::oak::log()->debug("404 {}", q.endpoint());
-    r.not_found(Json{
-        {"error", "cannot find handler"},
-        {"endpoints", Json(this->muxer.routes())},
+/**
+ * @brief Represents the http-connection
```

Comment:
Describing classes is where I would recommend the descriptive style. Sometimes there's not much to say and then I don't know what's better, no documentation, or repeating the obvious. One thing I will say, this documentation should be proper english. If we have to write this, try something like:
```
/**
 * Represents an incoming HTTP connection from a client to the server.
 */
```
That's more useful, I think.